src/System.Console/src/System/Console.cs

```diff
@@ -93,9 +93,8 @@ public static void ResetColor()
         // This is the worker delegate that is called on the Threadpool thread to fire the actual events. It sets the DelegateStarted flag so
         // the thread that queued the work to the threadpool knows it has started (since it does not want to block indefinitely on the task
         // to start).
-        private static void ControlCDelegate(object data)
+        private static void ControlCDelegate(this ControlCDelegateData controlCData)
```

Comment:
Any reason it can't just be an instance method, e.g. ControlCDelegateData.Run() or something like that, rather than a static extension method?